sympy/polys/series/ring.py

```diff
@@ -1,109 +1,1007 @@
+        x - 2/3*x**3 + 2/15*x**5 + O(x**6)
+
+        """
+        if not other:
+            return self
+
+        if isinstance(other, PowerSeriesElement):
+            return self._mul(other)
+
+        domain = self.ring.domain
+        if isinstance(other, int):
+            return self._mul_ground(self.ring.domain_new(other))
+        elif domain.of_type(other):
+            return self._mul_ground(other)
+        else:
+            raise NotImplementedError
+
+    def __rmul__(self, other: Er | int) -> PowerSeriesElement[Er]:
+        domain = self.ring.domain
+        if isinstance(other, int):
+            return self._mul_ground(self.ring.domain_new(other))
+        elif domain.of_type(other):
+            return self._mul_ground(other)
+        else:
+            raise NotImplementedError
+
+    def __pow__(self, n: int) -> PowerSeriesElement[Er]:
+        """
+        Raise a power Series to a non-negative integer power.
+
+        Examples
+        ========
+
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> (1 + x)**3
+        1 + 3*x + 3*x**2 + x**3
+
+        """
+        if n == 0:
+            return self.ring.one
+        if n == 1:
+            return self
+        else:
+            return self._pow_int(n)
+
+    def __truediv__(
+        self, other: PowerSeriesElement[Er] | Er | int
+    ) -> PowerSeriesElement[Er]:
+        """
+        Divide two power Series.
+
+        Examples
+        ========
+
+        >>> from sympy import QQ
+        >>> from sympy.polys.series import PowerSeriesRingField
+        >>> R = PowerSeriesRingField(QQ, "x")
+        >>> x = R.gen
+        >>> t = R.sin(x) / R.cos(x); t
+        x + 1/3*x**3 + 2/15*x**5 + O(x**6)
+        >>> R.tan(x) == t
+        True
+        """
+        if not other:
+            return self
+
+        if isinstance(other, PowerSeriesElement):
+            return self._div(other)
+
+        domain = self.ring.domain
+        if isinstance(other, int):
+            return self._div_ground(self.ring.domain_new(other))
+        elif domain.of_type(other):
+            return self._div_ground(other)
+        else:
+            raise NotImplementedError
+
+    def __rtruediv__(self, other: Er | int) -> PowerSeriesElement[Er]:
+        domain = self.ring.domain
+        if isinstance(other, int):
+            return self._rdiv_ground(self.ring.domain_new(other))
+        elif domain.of_type(other):
+            return self._rdiv_ground(other)
+        else:
+            raise NotImplementedError
+
+    def _add(self, other: PowerSeriesElement[Er]) -> PowerSeriesElement[Er]:
+        R = self.ring.ring
+        series = R.add(self.series, other.series)
+        return self.new(series)
+
+    def _add_ground(self, other: Er) -> PowerSeriesElement[Er]:
+        s2 = self.ring.ground_new(other)
+        return self._add(s2)
```

Comment:
Part of the point of methods like `add_ground` is not just to convert the type but to have a more efficient implementation. We should not need to create `s2` here.